presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -272,7 +278,11 @@ void PrestoServer::run() {
   std::unique_ptr<http::HttpsConfig> httpsConfig;
   if (httpsPort.has_value()) {
     folly::SocketAddress httpsSocketAddress;
-    httpsSocketAddress.setFromLocalPort(httpsPort.value());
+    if (bindToNodeInternalAddressOnly) {
+      httpSocketAddress.setFromHostPort(address_, httpsPort.value());
+    } else {
+      httpSocketAddress.setFromLocalPort(httpsPort.value());
```

Comment:
@arhimondr This needs to be `httpsSocketAddress` as below we create httpsConfig with it.